core/arch/arm/plat-zynqmp/drivers/puf.c

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2021 Foundries.io Ltd.
+ *
+ * Jorge Ramirez-Ortiz <jorge@foundries.io>
+ */
+
+#include <initcall.h>
+#include <io.h>
+#include <kernel/delay.h>
+#include <mm/core_memprot.h>
+
+#include "puf.h"
+
+/* CSU */
+#define ISR				0x20
+#define PUF_ISR_ACC_ERROR_MASK          BIT(12)
+
+/* PUF */
+#define CSU_PUF_BASE			(CSU_BASE + 0x4000)
```

Comment:
Base addresses would fit nicely in `platform_config.h`. (or even a separate register map header).

Reply:
yeah I wasnt sure. I opted for adding it here since I nobody else should care for that address